### PR TITLE
Fix llama tokenizer

### PR DESCRIPTION
**Pull request by ArthurZucker**
# What does this PR do?
Draft but:
- Fixes the conversion script 
- update the llama default special tokens 
- fixed compatibility issues
- cleanup llama tokeniztion code
- add tests

### CODE REVIEW

**Comment by ArthurZucker** [comment]
cc @Narsil for visibility!

**Comment by ArthurZucker** [comment]
This will need to wait for #22341

**Review by Narsil — COMMENTED** [review]
Let's put some tests before merging. (And a PR description)

Probably focusing on the breaking changes we're making here.

**Comment by ArthurZucker** [comment]
Yes, on it!

**Comment by ArthurZucker** [comment]
Will finish this tomorrow!

**Comment by Varal7** [comment]
Hi! Does this PR the decoding part of the tokenizer? Seems like it always prefixes the output with space.

For instance, `tokenizer.decode(1)` returns ` <s>`,

**Comment by ArthurZucker** [comment]
Yes, it does: `print(f'\'{tokenizer.decode(tokenizer.encode("Hello world"), skip_special_tokens = True)}\'',)` outputs `'Hello world'  😉

**Review by sgugger — APPROVED** [review]
Nice, thanks for all the fixes and for adding the tests!

**Review by Narsil — APPROVED** [review]
LGTM

File: tests/test_tokenization_common.py [hunk 1]
```diff
@@ -3932,7 +3932,7 @@ def test_clean_up_tokenization_spaces(self):
             tokenizer_fast.save_pretrained(tmp_dir_2)
             tokenizer = BertTokenizer.from_pretrained(tmp_dir_2)
 
-        assert tokenizer_fast.clean_up_tokenization_spaces is False
+        assert tokenizer.clean_up_tokenization_spaces is False
```
[the remark this answers is not in the document]
Comment (ArthurZucker) [reply tests/test_tokenization_common.py:3935]: this is not pointing to the correct part of the test. If the `cleanup_tokenization_spaces` is indeed False, the fail can happen for cache reasons or anything else  (also failed for me at some point). 
Will check again

**Pull request merged**